test/typetree.jl

```diff
@@ -38,4 +48,7 @@ end
     @test at2.y == 0.0
     @test at2.z == 0.0
     @test at2.type == 4
+
+    @test tt(LList2{Float64}) == "{[8]:Float@double}"
+    @test tt(Sibling) == "{[-1]:Pointer, [-1,8]:Float@double}"
```

Comment:
@wsmoses this looks wrong to me, but is the same answer that Enzyme currently gives.

Reply:
The reason it appears right is that unions should intersect the inner types, which here would be nothing I presume. This is because TT's don't represent unions but represent a guarantee that this offset is always type x